user/provider.go

```diff
@@ -35,10 +40,32 @@ func (p Provider) SaveUsers(users *[]User) error {
 	return helper.SaveJSONFile(FileName, users)
 }
 
-func (p Provider) SaveUser(user *User) error {
+func (p Provider) SaveUser(user *User, Id int) error {
+	users, err := p.GetAll()
+	if err!= nil {
+		return err
+	}
+
+	for _, usr := range users{
+	if usr.Id != Id{
+		helper.SaveJSONFile(FileName, user)
+		}
+	
+		jsonData, err := json.Marshal(user)
+		if err != nil {
+			return err
+		}
+	
+		err = os.WriteFile(FileName, jsonData, 0644)
```

Comment:
never save in a loop

Reply:
or save every 10/100/1000 times
if updatedCount > 100 {
  updatedCount = 0
  os.WriteFile...
}